src/lib.rs

```diff
@@ -467,10 +509,87 @@ where
     self.r_U_secondary = r_U_secondary;
     self.r_W_secondary = r_W_secondary;
 
+    self.ri_primary = r_next_primary;
+    self.ri_secondary = r_next_secondary;
+
     Ok(())
   }
 
-  /// Verify the correctness of the `RecursiveSNARK`
+  /// add a randomizing fold to a `RecursiveSNARK`
+  fn randomizing_fold(
```

Comment:
I wonder if this method could be inside `impl RecursiveSNARKRandomized` (if we apply the renaming per https://github.com/microsoft/Nova/pull/330/files#r1806864934)

Reply:
yeah - maybe this method is what makes a `RecursiveSNARKRandomized` object?